tests/dtm/test_multi_m0d_process_restart.py

```diff
@@ -0,0 +1,208 @@
+                cls.master_node_list.append(node_obj)
+            else:
+                cls.worker_node_list.append(node_obj)
+        cls.setup_type = CMN_CFG["setup_type"]
+        cls.test_completed = False
+        cls.health_obj = Health(cls.master_node_list[0].hostname,
+                                cls.master_node_list[0].username,
+                                cls.master_node_list[0].password)
+        cls.test_cfg = configmanager.get_config_wrapper(fpath="config/dtm/test_dtm_config.yaml")
+        cls.m0d_process = 'm0d'
+        cls.rgw_process = 'radosgw'
+        cls.log.info("Setup S3bench")
+        resp = s3bench.setup_s3bench()
+        assert_utils.assert_true(resp)
+        cls.ha_obj = HAK8s()
+        cls.rest_obj = S3AccountOperations()
+        cls.setup_type = CMN_CFG["setup_type"]
+        cls.system_random = secrets.SystemRandom()
+        cls.test_dir_path = os.path.join(TEST_DATA_FOLDER, "DTMTestData")
+        cls.delay = cls.test_cfg['delay']
+
+    def setup_method(self):
+        """Setup Method"""
+        self.log.info("Check the overall status of the cluster.")
+        resp = self.ha_obj.check_cluster_status(self.master_node_list[0])
+        assert_utils.assert_true(resp[0], resp[1])
+        self.log.info("Cluster status is online.")
+
+        self.random_time = int(perf_counter_ns())
+        self.s3acc_name = f"dps_s3acc_{self.random_time}"
+        self.s3acc_email = f"{self.s3acc_name}@seagate.com"
+        self.bucket_name = f"dps-bkt-{self.random_time}"
+        self.object_name = f"dps-obj-{self.random_time}"
+        self.deploy = False
+        self.iam_user = dict()
+        self.log.info("Create IAM user with name %s", self.s3acc_name)
+        resp = self.rest_obj.create_s3_account(acc_name=self.s3acc_name,
+                                               email_id=self.s3acc_email,
+                                               passwd=S3_CFG["CliConfig"]["s3_account"]["password"])
+        assert_utils.assert_true(resp[0], resp[1])
+        self.access_key = resp[1]["access_key"]
+        self.secret_key = resp[1]["secret_key"]
+        self.iam_user = {'s3_acc': {'accesskey': self.access_key, 'secretkey': self.secret_key,
+                                    'user_name': self.s3acc_name}}
+        self.s3_test_obj = S3TestLib(access_key=self.access_key, secret_key=self.secret_key,
+                                     endpoint_url=S3_CFG["s3_url"], max_attempts=0)
+        self.dtm_obj = DTMRecoveryTestLib(self.access_key, self.secret_key, max_attempts=0)
+        self.log.info("Created IAM user with name %s", self.s3acc_name)
+        if not os.path.exists(self.test_dir_path):
+            system_utils.make_dirs(self.test_dir_path)
+
+        self.log.info("Edit deployment file to append conditional sleep command to m0d setup "
+                      "command of all data pods.")
+        resp = self.master_node_list[0].get_deployment_name(POD_NAME_PREFIX)
```

Comment:
Now, we don't have deployments for server and data pods, we have statefulsets for both. So deployment file also won't be there. You will have to check other way of doing this.

Reply:
In progress for this.